toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -214,10 +214,17 @@ func fstabEntriesToMountPoints(fstabEntries []diskutils.FstabEntry, diskPartitio
 			return nil, nil, nil, err
 		}
 
-		// Unset read-only flag so that read-only partitions can be customized.
-		// Unset noexec flag so that if rootfs is set as noexec, image can still be customized. For example, allowing
-		// grub2-mkconfig to be called.
-		vfsOptions := fstabEntry.VfsOptions & ^diskutils.MountFlags(unix.MS_RDONLY|unix.MS_NOEXEC)
+		var vfsOptions diskutils.MountFlags
+		if overrideReadonly {
+			// Unset read-only flag so that read-only partitions can be customized.
+			// Unset noexec flag so that if rootfs is set as noexec, image can still be customized. For example, allowing
+			// grub2-mkconfig to be called.
+			vfsOptions = fstabEntry.VfsOptions & ^diskutils.MountFlags(unix.MS_RDONLY|unix.MS_NOEXEC)
+		} else {
+			// In post-customization scenarios (e.g. when the image is re-mounted for injection),
```

Comment:
We have `postCustomization` scripts. So, saying "post-customization" here might be a tad confusing.

Reply:
Fixed.